pkg/controllers/globalsettings/globalsettings.go

```diff
@@ -0,0 +1,78 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package globalsettings
+
+import (
+	"context"
+	"sync"
+
+	"github.com/avast/retry-go"
+	"github.com/samber/lo"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"knative.dev/pkg/logging"
+	"knative.dev/pkg/system"
+)
+
+const (
+	controllerName = "global-settings"
+	configMapName  = "karpenter-global-settings"
+)
+
+type settingsKeyType struct{}
+
+var SettingsKey = settingsKeyType{}
+
+type Settings struct {
+	mu   sync.RWMutex
+	data map[string]string
+}
```

Comment:
A neat pattern for this I learned recently. 
```
type Settings map[string]string

synchronizedSettings := make(chan Settings)

func() {
  settings <-synchronizedSettings
  // mutate settings
  synchronizedSettings <- settings
}
```

Reply:
This doesn't work well as a `RWMutex`, mainly because you don't know the number of readers. Maybe this works better as a standard `Mutex`?

Reply:
Ah -- good point.